services/dynamic-scheduler/tests/unit/status_monitor/test_services_status_monitor__monitor.py

```diff
@@ -0,0 +1,409 @@
+# pylint:disable=redefined-outer-name
+# pylint:disable=unused-argument
+
+import json
+import re
+from collections.abc import AsyncIterable, Callable
+from copy import deepcopy
+from typing import Any
+from unittest.mock import AsyncMock
+from uuid import uuid4
+
+import pytest
+import respx
+from fastapi import FastAPI, status
+from fastapi.encoders import jsonable_encoder
+from httpx import Request, Response
+from models_library.api_schemas_directorv2.dynamic_services import DynamicServiceGet
+from models_library.api_schemas_dynamic_scheduler.dynamic_services import (
+    DynamicServiceStart,
+    DynamicServiceStop,
+)
+from models_library.api_schemas_webserver.projects_nodes import NodeGet, NodeGetIdle
+from models_library.projects_nodes_io import NodeID
+from pydantic import NonNegativeInt
+from pytest_mock import MockerFixture
+from pytest_simcore.helpers.typing_env import EnvVarsDict
+from settings_library.rabbit import RabbitSettings
+from settings_library.redis import RedisSettings
+from simcore_service_dynamic_scheduler.services.service_tracker import (
+    get_all_tracked,
+    set_request_as_running,
+    set_request_as_stopped,
+)
+from simcore_service_dynamic_scheduler.services.status_monitor import _monitor
+from simcore_service_dynamic_scheduler.services.status_monitor._deferred_get_status import (
+    DeferredGetStatus,
+)
+from simcore_service_dynamic_scheduler.services.status_monitor._monitor import Monitor
+from simcore_service_dynamic_scheduler.services.status_monitor._setup import get_monitor
+from tenacity import AsyncRetrying
+from tenacity.retry import retry_if_exception_type
+from tenacity.stop import stop_after_delay
+from tenacity.wait import wait_fixed
+
+pytest_simcore_core_services_selection = [
+    "rabbit",
+    "redis",
+]
+
+
+@pytest.fixture
+def app_environment(
+    app_environment: EnvVarsDict,
+    rabbit_service: RabbitSettings,
+    redis_service: RedisSettings,
+    remove_redis_data: None,
+) -> EnvVarsDict:
+    return app_environment
+
+
+_DEFAULT_NODE_ID: NodeID = uuid4()
```

Comment:
there is a fixture for node ids

Reply:
yes but there is no way to use it outside of a test.